src/framework/analysisResult.ml

```diff
@@ -45,6 +45,18 @@ struct
     let defline () = dprintf "OTHERS -> Not available\n" in
     dprintf "@[Mapping {\n  @[%t%t@]}@]" content defline
 
+  let pretty_deterministic () mapping =
+    let bindings =
+      to_list mapping
+      |> List.sort [%ord: ResultNode.t * Range.t]
+    in
+    let f dok (key, st) =
+      dok ++ dprintf "%a ->@?  @[%a@]\n" ResultNode.pretty key Range.pretty st
+    in
+    let content () = List.fold_left f nil bindings in
+    let defline () = dprintf "OTHERS -> Not available\n" in
+    dprintf "@[Mapping {\n  @[%t%t@]}@]" content defline
+
```

Comment:
Can some code be reused between `pretty` and `pretty_deterministic` here?

Reply:
Theoretically yes, but I suspect it's not worth the trouble and it won't end up being shorter or simpler either. `BatHashtbl.fold` and `List.fold_left` are different signatures and `BatHashtbl` cannot be converted to `Seq.t` (#1443).